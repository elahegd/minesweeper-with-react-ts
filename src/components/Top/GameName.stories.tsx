import React from 'react';
import { Story, Meta } from '@storybook/react';

import { GameName } from './GameName';

export default {
    title: "Top/GameName",
    component: GameName
} as Meta;

const Template: Story = (args) => <GameName {...args} />;

export const MinesweeperGameName = Template.bind({});
MinesweeperGameName.args = {
    children: 'minesweeper'
}