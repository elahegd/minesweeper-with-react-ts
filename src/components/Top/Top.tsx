import { FC, memo } from 'react';
import styled from '@emotion/styled';

import { Legend, LegendProps } from './Legend';
import { GameName } from './GameName';

export interface TopProps extends LegendProps {
    children: string;
}

export const Top: FC<TopProps> = memo(({ children, ...legendProps }) => (
  <Header>
    <GameName>{children}</GameName>
    <Legend {...legendProps} />
  </Header>
));
// Stryker disable next-line StringLiteral
// Useful for debuging in devtools
Top.displayName = 'Top';

const Header = styled.header`
  text-align: center;
  position: relative;
  display: inline-block;
`;