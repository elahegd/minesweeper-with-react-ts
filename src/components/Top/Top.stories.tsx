import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Top, TopProps } from './Top';

export default {
    title: "Top/Top",
    component: Top
} as Meta;

const Template: Story<TopProps> = (args) => <Top {...args} />;

export const GameTop = Template.bind({});
GameTop.args = {
    children: 'minesweeper',
    feature: 'Flag',
    firstAction: 'ctrl',
    secondAction: 'click',
}