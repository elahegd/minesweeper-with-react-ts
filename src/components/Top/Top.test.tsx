import React from 'react';
import { render } from "@testing-library/react";

import { Top } from "./Top";

describe("Top renders coreectly", () => {
    it("get snapshot from Top component", () => {
        const { asFragment } = render(<Top feature={'Flag'}
        firstAction={'ctrl'}
        secondAction={'click'}>minesweeper</Top>);

        expect(asFragment()).toMatchSnapshot();
    })
})