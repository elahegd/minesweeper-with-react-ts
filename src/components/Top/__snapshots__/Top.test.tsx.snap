// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top renders coreectly get snapshot from Top component 1`] = `
<DocumentFragment>
  <header
    class="css-dn2347"
  >
    <h1
      class="css-18kvaxm"
    >
      minesweeper
    </h1>
    <legend
      class="css-17qz6a2"
    >
      <strong>
        Flag: 
      </strong>
      <code
        class="css-12ap3z0"
      >
        <span
          class="css-udfepp"
        >
          ctrl
        </span>
        <span
          class="css-9c23k1"
        >
           + click
        </span>
      </code>
    </legend>
  </header>
</DocumentFragment>
`;
